packages/common/src/directives/ng_optimized_image/image_loaders/netlify_loader.ts

```
@@ -0,0 +1,92 @@
+ * but it's a good start for sites in development. This is only used to warn users who haven't
+ * configured an image loader.
+ */
+function isNetlifyUrl(url: string): boolean {
+  return NETLIFY_LOADER_REGEX.test(url);
+}
+
+/**
+ * Function that generates an ImageLoader for Netlify and turns it into an Angular provider.
+ *
+ * @param path optional URL of the desired Netlify site. Defaults to the current site.
+ * @returns Set of providers to configure the Netlify loader.
+ *
+ * @publicApi
+ */
+export function provideNetlifyLoader(path?: string) {
+  if (path && !isValidPath(path)) {
+    throw new RuntimeError(
+      RuntimeErrorCode.INVALID_LOADER_ARGUMENTS,
+      ngDevMode &&
+        `Image loader has detected an invalid path (\`${path}\`). ` +
+          `To fix this, supply either the full URL to the Netlify site, or leave it empty to use the current site.`,
+    );
+  }
+
+  if (path) {
+    const url = new URL(path);
+    path = url.origin;
+  }
+
+  const loaderFn = (config: ImageLoaderConfig) => {
+    return createNetlifyUrl(path, config);
+  };
+
+  const providers: Provider[] = [{provide: IMAGE_LOADER, useValue: loaderFn}];
+  return providers;
+}
+
+const validParams = new Map<string, string>([
+  ['height', 'h'],
+  ['fit', 'fit'],
+  ['quality', 'q'],
+  ['q', 'q'],
+  ['position', 'position'],
+]);
+
+function createNetlifyUrl(path: string = 'https://a/', config: ImageLoaderConfig) {
+  const url = new URL(path);
+  url.pathname = '/.netlify/images';
+
+  if (!isAbsoluteUrl(config.src) && !config.src.startsWith('/')) {
+    config.src = '/' + config.src;
+  }
+
+  url.searchParams.set('url', config.src);
+
+  if (config.width) {
+    url.searchParams.set('w', config.width.toString());
+  }
+
+  for (const [param, value] of Object.entries(config.loaderParams ?? {})) {
+    if (validParams.has(param)) {
+      url.searchParams.set(validParams.get(param)!, value.toString());
+    }
```

It'd be great to have some code in dev mode to produce a console.warn in case an invalid parameter is provided (vs silently ignoring it), i.e. something like this:

```
    if (validParams.has(param)) {
      url.searchParams.set(validParams.get(param)!, value.toString());
    } else {
      if (ngDevMode) {
        console.warn(formatRuntimeError(RuntimeErrorCode.INVALID_LOADER_ARGUMENTS, '...'));
      }
    }
```